lib/current_user.rb

```diff
@@ -13,8 +13,8 @@ def clear_current_user
     @current_user_provider = Discourse.current_user_provider.new({})
   end
 
-  def log_on_user(user)
-    current_user_provider.log_on_user(user, session, cookies)
+  def log_on_user(user, opts)
```

Comment:
I think `opts` needs to be optional - that's what's causing all the specs to fail in travis.
```suggestion
  def log_on_user(user, opts={})
```